frontend/src/components/roster/View.js

```diff
@@ -0,0 +1,355 @@
+            })}
+          </span>{" "}
+          to{" "}
+          <span class="inline-flex font-medium">
+            {new Date(
+              Date.parse(tournamentQuery.data?.event?.registration_end_date)
+            ).toLocaleDateString("en-US", {
+              year: "numeric",
+              month: "short",
+              day: "numeric",
+              timeZone: "UTC"
+            })}
+          </span>
+        </Warning>
+      </div>
+
+      <div class="mx-auto max-w-screen-md">
+        <div class="mt-10">
+          <h4 class="text-xl font-bold text-blue-500">Add to Roster</h4>
+          <Show
+            when={tournamentQuery.data?.status == "REG"}
+            fallback={
+              <p class="mt-4 text-sm italic text-gray-500">
+                Registrations have closed !
+              </p>
+            }
+          >
+            <Switch>
+              <Match when={!store.loggedIn}>
+                <p class="mt-4 text-sm italic text-gray-500">
+                  You must be logged in to add/remove players from the roster !
+                </p>
+              </Match>
+              <Match
+                when={
+                  store.loggedIn &&
+                  (!teamQuery.data?.admins || !currentUserIsTeamAdmin())
+                }
+              >
+                <p class="mt-4 text-sm italic text-gray-500">
+                  You must be a team admin to perform rostering !
+                </p>
+                <p class="mt-4 text-sm italic text-gray-500">
+                  Your team admins are:
+                </p>
+              </Match>
+              <Match when={store.loggedIn && currentUserIsTeamAdmin()}>
+                <p class="mt-4 text-sm italic text-gray-500">
+                  Rostering here...
+                </p>
+              </Match>
+            </Switch>
+          </Show>
+        </div>
+        <div class="mt-12">
+          <h4 class="text-xl font-bold text-blue-500">Current Roster</h4>
+          <h2 class="mt-4 text-lg font-bold">Players</h2>
+          <Show
+            when={players()?.length !== 0}
+            fallback={
+              <p class="mt-4 text-sm italic text-gray-500">
+                0 players in roster yet...
+              </p>
+            }
+          >
+            <div
+              class={clsx("mt-4", currentUserIsTeamAdmin() ? "divide-y" : "")}
+            >
+              <For each={players()}>
+                {registration => (
+                  <div
+                    class={clsx(
+                      "mr-6 flex w-full items-center justify-between space-x-4 pr-2",
+                      currentUserIsTeamAdmin() ? "py-4" : "py-2"
+                    )}
+                  >
+                    <div class="flex items-center gap-x-4">
+                      <img
+                        class="h-10 w-10 rounded-full p-1 ring-2 ring-gray-300 dark:ring-gray-500"
+                        src={registration.person.image_url}
```

Comment:
I think we can remove is this will be null in case person doesnt exist or we should have a default image url then in that case